winml/lib/Api/ImageFeatureValue.cpp

```diff
@@ -502,43 +409,34 @@ HRESULT ImageFeatureValue::GetOrtValue(WinML::BindingContext& context, OrtValue*
   // Get the session
   auto spSession = context.session.as<LearningModelSession>();
   auto spDevice = spSession->Device().as<LearningModelDevice>();
-  auto provider = spSession->GetExecutionProvider();
-
-  // and the adapter
-  if (!m_adapter) {
-    WINML_THROW_IF_FAILED(OrtGetWinMLAdapter(m_adapter.put()));
-  }
+  auto engine = spSession->GetEngine();
 
   // create the OrtValue
-  Ort::Allocator dml_allocator(m_adapter.get(), nullptr);
-  WINML_THROW_IF_FAILED(m_adapter->GetProviderAllocator(provider, dml_allocator.put()));
-
-  // create the OrtValue as a tensor letting ort know that we own the data buffer
-  Ort::Value ort_tensor = Ort::Value::CreateTensor(
-      dml_allocator,
-      &(resourceMetadata.TensorDescriptor.sizes[0]),
-      sizeof(resourceMetadata.TensorDescriptor.sizes) / sizeof(resourceMetadata.TensorDescriptor.sizes[0]),
-      (resourceMetadata.TensorDescriptor.dataType == kImageTensorDataTypeFloat32) ? ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT : ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16);
+  winrt::com_ptr<IValue> value;
+  RETURN_IF_FAILED(engine->CreateTensorValue(
+	  resourceMetadata.TensorDescriptor.sizes,
+	  sizeof(resourceMetadata.TensorDescriptor.sizes) / sizeof(resourceMetadata.TensorDescriptor.sizes[0]),
+      resourceMetadata.TensorDescriptor.dataType == kImageTensorDataTypeFloat32 ? winml::TensorKind::Float : winml::TensorKind::Float16,
+	  value.put()));
 
   // Get the tensor raw data
-  void* pAllocatedResource = nullptr;
-  Ort::ThrowOnError(Ort::GetApi().GetTensorMutableData(ort_tensor, &pAllocatedResource));
+  auto allocated_resource = value->GetResource();
```

Comment:
>allocated_resource [](start = 7, length = 18)

not the "allocated resource" anymore, as the dml resource is hidden in the ort lib now - it is the actual resource...